### DISCUSSION TITLE
Hope weather apps save battery

### DISCUSSION THREAD

**Discussion by OSNewbie** [breezy-weather/breezy-weather#887] [Q&A] [answered]
Because I saw that the permissions have been increased, ignoring the battery optimization and self-starting permissions. I feel that these two permissions are not needed. These two permissions will increase the power consumption of the mobile phone. I hope to remove the permissions that consume the power of the mobile phone. Let’s look at the weather.  It just relies on the network permission to refresh weather data. If it is positioned in a region, just use the positioning permission. Are you sure it is based on network positioning or positioning permission?

**Comment by papjul** [comment, marked as answer]
Turning off battery optimization is optional. Note that this doesn't mean it will increase the power consumption, it just makes the app more reliable on non-conventional Android phones such as Samsung or Xiaomi. If you have a standard Android, you don't need to enable it. However, if you don't enable it on non-standard Android and you encounter errors, we won't investigate your issue.

Starting on boot permission is necessary again on non-conventional phones such as Xiaomi, because if not enabled, it will not resume background workers and forecast notifications on these phones, which is a violation of the Android standard. On standard Android phones, it does not increase power consumption as the app does nothing in particular when starting on boot. Note that background workers are efficient and only refresh the locations you actually use, and that you can turn it off from settings, although I would highly discourage you from doing that given how insignificant the power consumption/network data usage is.

As for the location permission, you can safely disable it and add your location manually from the search icon on your location list.
